### ISSUE TITLE
Color contrast should take styled THEAD/TFOOT into account

### ISSUE THREAD

**Issue by schne324** [dequelabs/axe-core#823]
on my y5e game http://harris-schneiderman.com/a11y-games/y5e/ I believe axe is raising a false positive for color contrast.  If you click "New game" and then run axe, it calls out each td in the thead as violating color contrast (it isn't properly calculating the background color).  Now I do realize that there is an unrelated issue I have in the table in which those tds should be ths but I figured I leave it as is to give you a chance to see the potential false positive.

NOTE: the extension states that the background color is `#dee8e6` when in fact it is `#0f635b`

**Comment by marcysutton** [comment]
Table elements like TH and TD had to be added manually to the color contrast algorithm (document.elementsFromPoint doesn't include those, along with other "edge" cases), and THEAD probably wasn't considered. As long as styling of THEAD is consistent in all browsers, we could add it to the algorithm. In the meantime, you could add the background color to your TH or TD elements instead as those are evaluated.

**Comment by schne324** [comment]
cool, thanks @marcysutton !! The solution here is for me to make the <td />s <th />s (because they should be and it was an initial oversight on my part) but I just wanted to point this weird result out before I make the fix on my end

**Comment by jeankaplansky** [comment]
No product docs required.

**Comment by somaalapati** [comment]
@WilcoFiers You closed and reopened this ticket. Did you close it accidentally? Or you need to fix this issue?

**Comment by somaalapati** [comment]
@chandana7393 Please test this ticket. Let me know if you need my help.

**Comment by chandana7393** [comment]
Tested, working as expected.
![tfoot_color_contrast](https://user-images.githubusercontent.com/45285854/62531933-0e22a200-b861-11e9-8267-2660b3c147aa.PNG)
![thead_color_contrast](https://user-images.githubusercontent.com/45285854/62531940-0fec6580-b861-11e9-8b29-1e2d808d2b58.PNG)
**Tested Environment:**
Attest - 2.5.1.21421v
Axe - 3.8.1.21421v
Axe-coconut - 3.8.1.21704v
Chrome - 76.0.3809.87v
Firefox - 68.0.1v
OS - Windows 10 64 bit.
